lib/hexpm/repository/assets.ex

```diff
@@ -92,36 +90,33 @@ defmodule Hexpm.Repository.Assets do
     Hexpm.Store.put(nil, :s3_bucket, docs_store_key(release), body, opts)
   end
 
-  defp delete_old_docs(release, paths, docs_for_latest_release) do
+  defp delete_old_docs(release, paths, publish_unversioned?) do
     name = release.package.name
     version = release.version
-    pre_release? = version.pre != []
 
     # Add "/" so that we don't get prefix matches, for example phoenix
     # would match phoenix_html
     existing_keys = Hexpm.Store.list(nil, :docs_bucket, "#{name}/")
-    keys_to_delete =
-      Enum.flat_map(existing_keys, fn key ->
-        first = Path.relative_to(key, name) |> Path.split |> hd
-        cond do
-          # Don't delete if we are going to overwrite with new files, this
-          # removes the downtime between a deleted and added page
-          key in paths ->
-            []
+    keys_to_delete = Enum.filter(existing_keys, &delete_key?(&1, paths, name, version, publish_unversioned?))
+    Hexpm.Store.delete_many(nil, :docs_bucket, keys_to_delete)
+  end
+
+  defp delete_key?(key, paths, name, version, publish_unversioned?) do
+    # Don't delete if we are going to overwrite with new files, this
+    # removes the downtime between a deleted and added page
+    if key in paths do
+      false
+    else
+      first = Path.relative_to(key, name) |> Path.split() |> hd()
+      case Version.parse(first) do
+        {:ok, first} ->
           # Current (/ecto/0.8.1/...)
-          first == version ->
```

Comment:
This is the fix, the comparison was: `"1.3.0" == #Version<1.3.0>`, changed to a proper `Version.compare/2.